codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java

```diff
@@ -332,7 +332,29 @@ public static void toHttp2Headers(HttpHeaders inHeaders, Http2Headers out) throw
                 // https://tools.ietf.org/html/rfc7540#section-8.1.2.2 makes a special exception for TE
                 if (!aName.contentEqualsIgnoreCase(HttpHeaderNames.TE) ||
                     AsciiString.contentEqualsIgnoreCase(entry.getValue(), HttpHeaderValues.TRAILERS)) {
-                    out.add(aName, AsciiString.of(entry.getValue()));
+                    if (aName.contentEqualsIgnoreCase(HttpHeaderNames.COOKIE)) {
+                        AsciiString value = AsciiString.of(entry.getValue());
+                        // split up cookies to allow for better compression
+                        // https://tools.ietf.org/html/rfc7540#section-8.1.2.5
+                        int index = value.forEachByte(FIND_SEMI_COLON);
+                        if (index != -1) {
+                            int start = 0;
+                            do {
+                                out.add(HttpHeaderNames.COOKIE, value.subSequence(start, index, false));
+                                start = index + 2; // skip 2 characters "; "
```

Comment:
I believe so. Let me add a reference to the cookie RFC to clarify.

https://tools.ietf.org/html/rfc6265#section-4.2.1

> cookie-header = "Cookie:" OWS cookie-string OWS
   cookie-string = cookie-pair *( ";" SP cookie-pair )

https://tools.ietf.org/html/rfc6265#section-4.1.1

> cookie-pair       = cookie-name "=" cookie-value